packages/react-reconciler/src/ReactFiberCommitWork.new.js

```diff
@@ -2250,22 +2250,25 @@ function commitMutationEffectsOnFiber(
     }
   }
 
-  if (flags & Visibility) {
-    switch (finishedWork.tag) {
-      case SuspenseComponent: {
-        const newState: OffscreenState | null = finishedWork.memoizedState;
+  switch (finishedWork.tag) {
+    case SuspenseComponent: {
+      const offscreenFiber: Fiber = (finishedWork.child: any);
```

Comment:
I checked the code base `Visibility` flag is only applied to `OffscreenComponent` or `LegacyHiddenComponent`,
so actually original code here is never run? which results in `markCommitTimeOfFallback()` not called 
and  the throttling failed to work.

so here I changed it to `finishedWork.child` to check the Visibility.

Or I can remove this `case SuspenseComponent`, but checking `finishedWork.return` in `OffscreenComponent` instead ?

Reply:
Is there a guarantee that `SuspenseComponent` has a `child` in all possible states?

Reply:
There are a few lines of existing code already assuming this should be always the case , for example 

https://github.com/facebook/react/blob/main/packages/react-reconciler/src/ReactFiberThrow.new.js#L345-L346
https://github.com/facebook/react/blob/main/packages/react-reconciler/src/ReactFiberCompleteWork.new.js#L1153

Also in the reconciler, [mountSuspenseFallbackChildren()](https://github.com/facebook/react/blob/main/packages/react-reconciler/src/ReactFiberBeginWork.new.js#L2343) and [mountSuspensePrimaryChildren()](https://github.com/facebook/react/blob/main/packages/react-reconciler/src/ReactFiberBeginWork.new.js#L2343) are both using the Offscreen fiber.

## But I agree with you,  we should invariant this.

I can fix all the places include the existing places, maybe better in a separate PR, how do you think?

Reply:
No, it's better not to add runtime checks if we're confident cause they slow things down. I was just double-checking.

Reply:
Got it. 
I wish there could be better typings rather than `any` here.